### ISSUE TITLE
only show notification achievement progress on some node

### ISSUE THREAD

**Issue by gameclamp** [Detanup01/gbe_fork#45]
Now we can just choice show notification or not show notification for achievement progress.
I know the game call the notification everytime we got progress.
I suggest show notification only when the new progress exceeds the previous one by 10% or 20%.Than we can keep this feature without being overwhelmed by too many notifications.

**Comment by Rustbeard86** [comment]
In _**configs.overlay.ini**_

`disable_achievement_progress=1` 

See **_configs.overlay.EXAMPLE.ini_** in \release\steam_settings.EXAMPLE\configs.overlay.EXAMPLE.ini for examples.

I realize this doesn't solve the issue, but because others may read this and wonder how to work around the spam notification that we see currently, I am putting this here.

**Comment by Detanup01** [comment]
No idea if the PR would be viable for you, please check.
If not I drop it, if you want this feature i add into it

**Comment by gameclamp** [comment]
> No idea if the PR would be viable for you, please check. If not I drop it, if you want this feature i add into it

Thank you very much.But it seem the code logic not true.If I set min_diff_progress = 3.When kill a zombie in game,the achievement progress not increase anymore.
Should it be this? :
1.add a new property to save the progress value last notified. e.g:last_notified_progress
2.set min_diff_progress in float like 0.1.
3.every time the game call the notification, check the ratio like this: (progress - last_notified_progress) / max_progress >=min_diff_progress ?

code run:
1: kill first zb. last_notified_progress==0? ~ true. show notification . last_notified_progress=progress=1
2: kill second zb. (2- 1) / 5000 >=0.1? ~ false. last_notified_progress: 1,progress: 2
3: kill third zb. (3- 1) / 5000 >=0.1? ~ false. last_notified_progress: 1,progress: 3
......
500: kill 500th zb. (500-1) / 5000 >=0.1? ~false. last_notified_progress: 1,progress: 500
501: kill 501th zb. (501-1) / 5000 >=0.1? ~ true. show notification. last_notified_progress: 501=progress=501
502: kill 500th zb. (502-501) / 5000 >=0.1? ~false. last_notified_progress: 501,progress: 502
.....

**Comment by Detanup01** [comment]
are you sure the % one is good?
Maybe something like `(progress - last_notified) >= min_diff`
(Set notif every 5)
5 - 0 = 5 | 5 >= 5 , set last_not to 5
6 - 5 = 1 | 1 >= 5 , nothing to do.

for floats:
set notif for 0.5
0.1 - 0 = 0.1 | 0.1 >= 0.5 | skip
0.5 -0 = 0.5 | 0.5 >= 05. set last notif to 0.5
0.6 - 0.5 = 0.1 | 0.1 >= 0.5 | skip

**Comment by gameclamp** [comment]
Because some games have big differences in it's achievement values.
e.g:
kill 20 boss
obtain 100000000000 gold
I think the % one is better in this situation.

**Comment by Detanup01** [comment]
Hmm
Yeah sure!
Doing % based after this.
Thanks!

**Comment by Detanup01** [comment]
Updated the PR please test the dll

**Comment by gameclamp** [comment]
Thank you for the update.
It seems there is no change in the behaviour.And the user's achievements.json didn't save the last_notified_progress.

**Comment by Detanup01** [comment]
sorry but have 0 idea where is it saving and how would i add that.
It works while you in game i think

**Comment by gameclamp** [comment]
I download the file emu-win-api_experimental-release-x64-42a5c18c888b219c776d8390e328c45362bfc503.zip，and set `stat_min_diff_progress=0.1` in `configs.main.ini` `[main::stats]`.
But the notification still show everytime.
![20241001225812](https://github.com/user-attachments/assets/11d6805c-bc14-46ad-8c4c-a4258fc3e090)

**Comment by mixa-pulemet** [comment]
I also encountered the problem of constant spam notifications about progress updates when playing Megabank. I would like to have some kind of setting for the frequency of progress notifications. Are there any plans to add this feature?

**Comment by GogoVang** [comment]
> I also encountered the problem of constant spam notifications about progress updates when playing Megabank. I would like to have some kind of setting for the frequency of progress notifications. Are there any plans to add this feature?

Just disable notifications all together and and they gonna appear only when game calls IndicateAchievementProgress, which is how it works on legit steam client.

**Comment by mixa-pulemet** [comment]
> > I also encountered the problem of constant spam notifications about progress updates when playing Megabank. I would like to have some kind of setting for the frequency of progress notifications. Are there any plans to add this feature?
> 
> Just disable notifications all together and and they gonna appear only when game calls IndicateAchievementProgress, which is how it works on legit steam client.

Thanks for the answer. If I understand correctly, if I set **disable_achievement_progress** to **1** in the **configs.overlay** file, and the game implements an **IndicateAchievementProgress** call, progress notifications will still be shown as programmed by the game developers, correct?

**Comment by GogoVang** [comment]
> Thanks for the answer. If I understand correctly, if I set **disable_achievement_progress** to **1** in the **configs.overlay** file, and the game implements an **IndicateAchievementProgress** call, progress notifications will still be shown as programmed by the game developers, correct?

No, you need to set stat_achievement_progress_functionality to 0 in configs.main.ini. Not a lot of games have this calls though, here's some from the top of my head if you wanna test it: 418240, 214490, 509580, 10680

**Comment by mixa-pulemet** [comment]
> No, you need to set stat_achievement_progress_functionality to 0 in configs.main.ini. Not a lot of games have this calls though, here's some from the top of my head if you wanna test it: 418240, 214490, 509580, 10680

Got it, thanks. Since the emulator already has a feature for displaying progress notifications, it would be nice to have a setting that limits the frequency of notifications. For example, for a game that uses a lot of frequently updated stats, you could set it so that notifications are displayed every 10% of the maximum stats value.
